tests/test_work_unit.py

```diff
@@ -343,6 +373,41 @@ def test_save_and_load_fits_shard(self):
             # We succeed if overwrite=True
             work.to_sharded_fits("test_workunit.fits", dir_name, overwrite=True)
 
+    def test_save_and_load_fits_shard_compressed(self):
+        with tempfile.TemporaryDirectory() as dir_name:
+            file_path = os.path.join(dir_name, "test_workunit.bz2")
+
+            # Write out the existing WorkUnit with a different per-image wcs for all the entries.
+            # work = WorkUnit(self.im_stack, self.config, None, self.diff_wcs)
+            work = WorkUnit(im_stack=self.im_stack, config=self.config, wcs=None, per_image_wcs=self.diff_wcs)
+            work.to_sharded_fits("test_workunit.bz2", dir_name)
+            self.assertTrue(Path(file_path).is_file())
+
+            # Read in the file and check that the values agree.
+            work2 = WorkUnit.from_sharded_fits(filename="test_workunit.bz2", directory=dir_name)
```

Comment:
for some reason my intuition would be that this wouldn't work with a sharded work unit, nice to know that it does !